### PR TITLE
Changeset merge

### PR DESCRIPTION
**Pull request by renfredxh**
Add Merge method to ChangesetServer and accompanying options

### CODE REVIEW

File: sourcegraph/sourcegraph.proto [hunk 1]
```diff
@@ -808,6 +808,8 @@ service Changesets {
 	// update event. If no update occurred, it returns nil.
 	rpc Update(ChangesetUpdateOp) returns (ChangesetEvent);
 
+	rpc Merge(ChangesetMergeOp) returns (pbtypes.Void);
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: My style guide and consistency advisor may complain that this is the only method within `Changesets` service that isn't documented.

File: sourcegraph/sourcegraph.proto [hunk 2]
```diff
@@ -986,6 +988,22 @@ message ChangesetUpdateOp {
 	UserSpec Author = 8 [(gogoproto.nullable) = false];
 }
 
+message ChangesetMergeOp {
+	// Repo holds the RepoSpec where the Changeset to be merged is located.
+	RepoSpec repo = 1 [(gogoproto.nullable) = false];
+
+	// ID holds the ID of the changeset that is to be merged.
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Suggestion, shorter comment form. Replace "X holds X ..." with juts "X ...". E.g., this comment can be:

```Go
// ID of the changeset that is to be merged.
```

File: sourcegraph/sourcegraph.proto [hunk 3]
```diff
@@ -986,6 +988,22 @@ message ChangesetUpdateOp {
 	UserSpec Author = 8 [(gogoproto.nullable) = false];
 }
 
+message ChangesetMergeOp {
+	// Repo holds the RepoSpec where the Changeset to be merged is located.
+	RepoSpec repo = 1 [(gogoproto.nullable) = false];
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Similarly, this could be:

```Go
// Repo of the the Changeset to be merged.
```

**Comment by shurcooL** [comment]
Left some minor documentation style suggestions. LGTM!

File: sourcegraph/sourcegraph.proto (hunk 1 above)
Comment (renfredxh) [inline sourcegraph/sourcegraph.proto]: :stuck_out_tongue:

File: sourcegraph/sourcegraph.proto (hunk 2 above)
Comment (renfredxh) [inline sourcegraph/sourcegraph.proto]: I agree, however this seems to be the default style (23 matches for "holds the" and 6 matches for "contains the") so I matched it for consistency

**Comment by renfredxh** [comment]
Manually merged

**Pull request closed without merging**